.github/workflows/publish-docs-preview.yml

```diff
@@ -11,15 +11,40 @@ jobs:
     if: "${{ github.event.label.name == ':eyes: publish-docs-preview' }}"
     steps:
 
+      - name: Find CI workflow run for PR
+        id: find-workflow-run
+        uses: actions/github-script@v7
+        continue-on-error: true
+        with:
+          script: |
+            // Find the last successful workflow run for the current PR's head
+            const {owner, repo} = context.repo;
+            const runsResponse = await github.rest.actions.listWorkflowRuns({
+              owner,
+              repo,
+              workflow_id: 'ci.yml',
+              event: 'pull_request',
+              head_sha: context.sha,
+              status: 'success'
+            });
+            const runs = runsResponse.data.workflow_runs;
+            runs.sort((a, b) => b.created_at.getTime() - a.created_at.getTime());
+            const run = runs[0];
+            if (!run) {
+              core.setFailed('Could not find a successful workflow run for the PR');
+              return;
+            }
+            core.setOutput('run-id', run.id);
+
       - name: Download artifact
+        if: ${{ steps.find-workflow-run.outcome == 'success' }}
         id: download-artifact
-        uses: dawidd6/action-download-artifact@v3
+        uses: actions/download-artifact@v4
         continue-on-error: true

```

Comment:
I'm dense. Why continue-on-error? If we fail to download artifact, why not let the error fly?

Reply:
Because we want to post a comment to the PR, with information about the failure. We check if the step was successful in subsequent steps.

Reply:
AH!